libr/anal/p/anal_arm_cs.c

```diff
@@ -1801,6 +1801,12 @@ jmp $$ + 4 + ( [delta] * 2 )
 			op->fail = addr + op->size;
 		}
 		break;
+	case ARM_INS_ADR:
+		op->type = R_ANAL_OP_TYPE_LEA;
+		// Set the pointer address and align it
+		op->ptr = IMM(1) + addr + 4 - (addr%4);
```

Comment:
In the 'Thumb ADR 16-bit encoding' line of Table 10-4 Register-relative offsets at  http://www.keil.com/support/man/docs/armasm/armasm_dom1361289862147.htm#/dom1361289862147__adr_reg_mult_fn it has a footnote that the offset range must be a multiple of 4.

I also found in the firmware I was analysing that all strings were aligned to addresses ending in 0,4,8,c so by subtracting addr%4 it always aligned and all strings resolved correctly.